CONTRIBUTING.rst

```diff
@@ -100,3 +73,51 @@ The full list of Ansible email lists and IRC channels can be found in the `commu
 .. _`freenode`: https://freenode.net
 .. _`molecule-users Forum`: https://groups.google.com/forum/#!forum/molecule-users
 .. _`communication page`: https://docs.ansible.com/ansible/latest/community/communication.html
+
+.. _community working group: https://github.com/ansible/community/tree/master/group-molecule
+
+Contribution Guidelines
+=======================
+
+* We are interested in various different kinds of improvement for Molecule;
+  please feel free to raise an `Issue`_ if you would like to work on something
+  major to ensure efficient collaboration and avoid duplicate effort.
+* Create a topic branch from where you want to base your work.
+* Check for unnecessary whitespace with ``git diff --check`` before committing.
+* Make sure you have added tests for your changes.
+* You must use ``git commit --signoff`` for any commit to be merged, and agree
+  that usage of ``--signoff`` constitutes agreement with the terms of `DCO 1.1`_.
+
+* Run all the tests to ensure nothing else was accidentally broken.
+* Reformat the code by following the formatting section below.
+* Submit a pull request.
+
+.. _`Issue`: https://github.com/ansible/molecule/issues
+.. _`DCO 1.1`: https://github.com/ansible/molecule/blob/master/DCO_1_1.md
+
+Installing
+==========
+
+:ref:`installation` from source or package.
+
+Testing
+=======
+
+Perform all of the :ref:`full_testing` testing steps prior to submitting a PR.
```

Comment:
I think this needs some further explanation, like: "because tests are heavy and unnecessary builds poison CI queue causing also a very long feedback cycle"